GeoLib/Polygon.cpp

```diff
@@ -417,27 +417,25 @@ void Polygon::splitPolygonAtPoint (std::list<GeoLib::Polygon*>::iterator polygon
 				std::swap (idx0, idx1);
 
 			// create two closed polylines
-			GeoLib::Polygon* polygon0 (new GeoLib::Polygon(*(*polygon_it)));
+			GeoLib::Polyline polyline0{*(*polygon_it)};
 			for (std::size_t k(0); k <= idx0; k++)
-				polygon0->addPoint ((*polygon_it)->getPointID (k));
-			for (std::size_t k(idx1 + 1); k < (*polygon_it)->getNumberOfPoints(); k++)
-				polygon0->addPoint ((*polygon_it)->getPointID (k));
-			polygon0->initialise();
+				polyline0.addPoint((*polygon_it)->getPointID(k));
+			for (std::size_t k(idx1 + 1);
+			     k < (*polygon_it)->getNumberOfPoints(); k++)
+				polyline0.addPoint((*polygon_it)->getPointID(k));
 
-			GeoLib::Polygon* polygon1 (new GeoLib::Polygon(*(*polygon_it)));
+			GeoLib::Polyline polyline1{*(*polygon_it)};
 			for (std::size_t k(idx0); k <= idx1; k++)
-				polygon1->addPoint ((*polygon_it)->getPointID (k));
-			polygon1->initialise();
+				polyline1.addPoint((*polygon_it)->getPointID(k));
 
 			// remove original polygon and add two new polygons
-			std::list<GeoLib::Polygon*>::iterator polygon0_it, polygon1_it;
-			polygon1_it =
-			        _simple_polygon_list.insert (_simple_polygon_list.erase (
-			                                             polygon_it), polygon1);
-			polygon0_it = _simple_polygon_list.insert (polygon1_it, polygon0);
-
-			splitPolygonAtPoint (polygon0_it);
-			splitPolygonAtPoint (polygon1_it);
+			auto polygon1_it = _simple_polygon_list.insert(
+			    _simple_polygon_list.erase(polygon_it), new Polygon(polyline1));
```

Comment:
You erase a pointer to a polygon from `_simple_polygon_list`. Does that leak memory?
Just a note: It seems like `*polygon_it = new Polygon(polyline1);` is equivalent to the given statement.